akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolConductor.scala

```diff
@@ -85,9 +90,12 @@ private object PoolConductor {
       Ports(retryMerge.in(0), retrySplit.in, route.outArray.toList)
     }
 
-  private case class SwitchCommand(rc: RequestContext, slotIx: Int)
+  sealed trait SlotCommand
+  case class DispatchCommand(rc: RequestContext) extends SlotCommand
+  case class SwitchSlotCommand(cmd: DispatchCommand, slotIx: Int) extends SlotCommand
+  case class SlotShouldConnectCommand(slotIx: Int) extends SlotCommand
```

Comment:
If I understand correctly, You would change
`SwitchSlotCommand(cmd: DispatchCommand, slotIx: Int)` into `SwitchSlotCommand(cmd: SlotCommand, slotIx: Int)`, right? 
The reason why I didn't go that way was that it, makes (for instance) embedding a `SwitchSlotCommand` in other `SwitchSlotCommand` possible. It's valid from the compilers point of view, but it makes little sense from the logical standpoint. 
What do you think?